README.md

```diff
@@ -1,11 +1,127 @@
+- [Introduction](#introduction)
+- [Installation of the virtual robot controller](#installation-of-the-virtual-robot-controller)
+  - [Virtual machine method](#virtual-machine-method)
+  - [Docker method](#docker-method)
+    - [Install Docker on Windows](#install-docker-on-windows)
+    - [Install Docker on macOS](#install-docker-on-macos)
+    - [Install Docker on Linux](#install-docker-on-linux)
+  - [RoboDK installation](#robodk-installation)
+- [Move and program the ursim virtual controller](#move-and-program-the-ursim-virtual-controller)
+- [Move and program the RoboDK virtual twin](#move-and-program-the-robodk-virtual-twin)
+- [Appendix](#appendix)
+  - [File provided for the Docker installation method](#file-provided-for-the-docker-installation-method)
+- [Additional features](#additional-features)
+- [References](#references)
+
+
+## Introduction
+
+Simulating a robot and performing off-line programming are crucial aspects to de-risk the success of your application. It allows you to test and validate your robot's behaviour in a virtual environment, reducing the time to debug and detect errors, which can take a long time on a real robot!
+
+This repository contains several tools and instructions on how to use Universal Robots (UR) offline simulator (ursim), a virtual controller replica of a real robot brain, capable of replicating the behaviour of a real robot one-to-one, with all features and programming capabilities of a real robot, minus the physical body of the arm!
+
+<!-- omit from toc -->
+###### Disclaimer
+
+There are a few ways to use this repository, with increasing level of difficulty to setup the virtual environments.
+The default method of installation suggests to use Docker, a more advanced software development environment setup, and a very useful and important tool to learn.
+
+<!-- omit from toc -->
+###### How to use this documentation
+
+- The documentation here will not give you an introduction to robotics, but it will show you the minimum required to:
+  - Install the virtual robot controller (two methods proposed).
+  - Move and program the virtual robot from the virtual pendant.
+  - Move and program a virtual twin from RoboDK.
+  - Connect the virtual twin and virtual controller to test the RoboDK programs.
+
+## Installation of the virtual robot controller
+
+Two different installation methods are provided, using virtual machines or Docker. Both can work on any host machine, whether it is macOS, Linux or Windows.
+
+> ⚠️ **Warning:** <br> Unfortunately, you will need an admin/root account access on your computer to make the docker installation work. So, in case you don't, please use the virtual machine installation method!
+
+### Virtual machine method
+
+- **Download Virtualbox VM:**
```

Comment:
In my experience, VirtualBox is much, *much* slower than VMware (or something like Parallels).

VMware and VMware Fusion are free for personal use these days, so might be options to consider.